fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingByIndex.java

```diff
@@ -96,64 +122,46 @@
                             .thenCompose(vignore -> {
                                 SubspaceProvider subspaceProvider = common.getRecordStoreBuilder().getSubspaceProvider();
                                 return subspaceProvider.getSubspaceAsync(context)
-                                        .thenCompose(subspace ->
-                                                buildIndexFromIndex(subspaceProvider, subspace)
-                                                        .thenCompose(vignore2 -> markBuilt(subspace)
-                                                        ));
+                                        .thenCompose(subspace -> buildIndexFromIndex(subspaceProvider, subspace, null, null));
                             });
                 }));
     }
 
-    @Nonnull
-    private CompletableFuture<Void> markBuilt(Subspace subspace) {
-        // Grand Finale - after fully building the index, remove all missing ranges in one gulp
-        return getRunner().runAsync(context -> {
-            final Index index = common.getIndex();
-            RangeSet rangeSet = new RangeSet(subspace.subspace(Tuple.from(FDBRecordStore.INDEX_RANGE_SPACE_KEY, index.getSubspaceKey())));
-            TransactionContext tc = context.ensureActive();
-            return rangeSet.insertRange(tc, null, null).thenApply(bignore -> null);
-        });
-    }
-
-    private void maybeLogBuildProgress(SubspaceProvider subspaceProvider, byte[] retCont) {
+    private void maybeLogBuildProgress(SubspaceProvider subspaceProvider) {
```

Comment:
I don't see this is unique to `IndexingByIndex`.  Maybe it's fine because you are going to get rid of is after finishing the full refactoring.